gammapy/spectrum/spectrum_grouping.py

```diff
@@ -48,17 +47,21 @@ def group_obs_table(obs_table, offset_range=[0, 2.5], n_off_bin=5,
     offmin, offmax = offset_range
     effmin, effmax = eff_range
     zenmin, zenmax = zen_range
-
+    coszenmin = np.cos(zenmax * np.pi / 180.)
+    coszenmax = np.cos(zenmin * np.pi / 180.)
     offtab = Angle(np.linspace(offmin, offmax, n_off_bin + 1), 'deg')
-    efftab = Quantity(np.linspace(effmin, effmax, n_eff_bin + 1)/100., '')
-    coszentab = Quantity(np.linspace(zenmin, zenmax, n_zen_bin + 1), '')
+    efftab = Quantity(np.linspace(effmin, effmax, n_eff_bin + 1) / 100., '')
+    coszentab = Quantity(np.linspace(coszenmin, coszenmax, n_zen_bin + 1), '')
 
```

Comment:
You should always use ``astropy`` when possible in ``gammapy`` (since it is an astropy affiliated package).
These line would then look like this
```
zentab = Quantity(np.linspace(zenmin, zenmax, n_zen_bin + 1), 'deg')
coszentab = np.cos(zentab)
```